Photon Transport Simulation with Two Layers of Material & Unstructured Mesh

anu1217 opened the pull request:
- Performing OpenMC photon transport using photon source generated by R2S Step2
  - Source_Mesh_Reader extracts source density data from R2S-generated meshes 
  - OpenMC_PhotonTransport contains main simulation
  - Photon_Tallytovtk reads data from statepoint file

This model currently uses the relative source intensities between different photon groups as the source strengths (and samples equally between individual mesh elements for any given energy group). However, it should be possible to add spatial variation of source intensity onto the MeshSpatial distribution.

anu1217 committed:
Add files via upload

anu1217 committed:
Deleted unused Energy_List lines

gonuke commented on OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,201 @@
+#Performing simulation for the first decay time
+with open('Sum_1.txt', 'r') as Sum_1:
+    Lines = Sum_1.readlines()
+    Split = Lines[0].split()
+    for sums in Split:
+        Strengths.append(float(sums))
```
```suggestion
    strengths = [float(strength) for strength in Lines[0].split()]
```

gonuke commented on OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,201 @@
+    Source_List = []
+    Particle_Filter = openmc.ParticleFilter('photon')
+    Total_Mesh = openmc.UnstructuredMesh(Mesh_File, library='moab')
+    Mesh_Dist = openmc.stats.MeshSpatial(Total_Mesh, volume_normalized=False)  
```
Move this into the loop and set the source strengths from the data you've extracted elsewhere

gonuke commented on OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,201 @@
+# Create materials & export to XML:
+#Simulating tungsten shell:
+  
+def make_W(element_1):
```
Move the geometry & materials functions into a separate python file that you can import into both neutron & photon transport

gonuke reviewed (changes requested):
Glad to see this is all (mostly) working

anu1217 committed:
Update OpenMC_PhotonTransport.py

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Delete unused 'Energy_List.txt' line

anu1217 committed:
Delete print statement

anu1217 committed:
Remove summed strengths text file as output

anu1217 committed:
Read summed photon strengths from Source_Mesh_Reader

- Reading total strengths directly from Source_Mesh_Reader instead of writing a new list in this script

anu1217 committed:
Add source strengths for each individual mesh element

- Brought the MeshBase object inside the loop over the energy bounds

anu1217 committed:
Update OpenMC_PhotonTransport.py

Assign material library path to new function and delete unused lines

anu1217 committed:
Adding materials/geometry files

These two files can be imported into a neutron or photon transport problem

anu1217 committed:
Import materials/geometry from TwoLayers_Materials & TwoLayers_Geometry

anu1217 committed:
Delete materials and geometry functions from current script

gonuke commented on TwoLayers_Geometry.py:
```diff
@@ -0,0 +1,33 @@
+
+# Create geometry
+#Spherical shell:
+def make_spherical_shell(inner_radius_W, outer_radius_W, inner_radius_C, M_1, M_2):    
```
docstrings here would help be clear about what everything is doing

gonuke commented on TwoLayers_Geometry.py:
```diff
@@ -0,0 +1,33 @@
+    Cells = [Void, W_Shell, C_Shell]
+    geometry = openmc.Geometry(Cells)
+    geometry.export_to_xml()
+    return geometry, Void, W_Shell, C_Shell, Cells
```
seems like you are returning more than necessary since `Cells` contains `W_Shell` and `C_Shell`, and `geometry` contains `Cells`

gonuke commented on TwoLayers_Geometry.py:
```diff
@@ -0,0 +1,33 @@
+    C_Shell = openmc.Cell(fill=M_2, region=S_C_3)
+    Cells = [Void, W_Shell, C_Shell]
+    geometry = openmc.Geometry(Cells)
+    geometry.export_to_xml()
```
Probably don't want to export here - separation of concerns points to putting `computing` and `input/output` in different methods.

gonuke commented on TwoLayers_Geometry.py:
```diff
@@ -0,0 +1,33 @@
+@author: Anupama Rajendra
+"""
+
+import openmc
```
Not clear that we need to import `openmc` here, if this method is only ever used when importing into another script that also imports `openmc`

anu1217 replied:
I might be missing something here, but OpenMC_PhotonTransport attempts to access openmc inside TwoLayers_Geometry even when only make_spherical_shells is imported

gonuke commented on TwoLayers_Geometry.py:
```diff
@@ -0,0 +1,33 @@
+    S_W_2 = openmc.Sphere(r=outer_radius_W, boundary_type='vacuum')
+    inside_W_sphere_2 = -S_W_2
+    outside_W_sphere_2 = +S_W_2
+    S_W_3 = outside_W_sphere_1 & inside_W_sphere_2 #filled with specified material
```
`S_W_1` and `S_W_2` are surfaces, but `S_W_3` is a region - better variable names will make this more readable.

It's also not clear that you need to define the intermediate variables `inside_...` and `outside_...`.  If `S_W_1` and `S_W_2` (and `S_C_1` below, etc) were named better, then the OpenMC notation with `-` and `+` might just be clear
```suggestion
    inner_W_sphere= openmc.Sphere(r=inner_radius_W) #sphere of radius 1000cm
    outer_W_sphere = openmc.Sphere(r=outer_radius_W, boundary_type='vacuum')
    W_region = +inner_W_sphere & -outer_W_sphere
```

gonuke commented on TwoLayers_Materials.py:
```diff
@@ -0,0 +1,41 @@
+            Density_M_2 = float(line.strip().split()[3])
+            M_2.set_density('g/cm3', Density_M_2)
+    M_2.add_element(element_2, 1.00)
+    return M_2
```
These two methods are identical, why two copies?  It looks like you just need to pass the `material_id` as another argument?

gonuke commented on TwoLayers_Materials.py:
```diff
@@ -0,0 +1,41 @@
+def mat_lib(filepath):  
+    with open(""+filepath+"") as ALARA_Lib:
+    #with open("elelib.std") as ALARA_Lib:
+        Lib_Lines = ALARA_Lib.readlines()
```
Looking at the way you use this data below, I recommend processing when you read it into a convenient form.  For example, you might want to make a dictionary of `element_densities` where the keys are the element symbol.

gonuke commented on TwoLayers_Materials.py:
```diff
@@ -0,0 +1,41 @@
+def all_mat(M_1, M_2):
+    all_materials = openmc.Materials([M_1, M_2])
+    all_materials.cross_sections = '../fendl-3.2-hdf5/cross_sections.xml'
+    all_materials.export_to_xml()
```
instead of exporting, return the `all_materials` object

gonuke commented on Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,41 @@
+# Process each file
+for file_index, file in enumerate(Files):
+    # Extract data from the current file
+    tstt = file['tstt']
+    elements = tstt['elements']
+    tet_four = elements['Tet4']
+    tags = tet_four['tags']
+    sd = tags['source_density'][:]
+    
+    # Write source density to a separate file for each mesh
+    with open(f'source_density_{file_index + 1}.txt', 'w') as source:
+        for tet_element in sd:
+            source.write(' '.join(map(str, tet_element)) + '\n')
+
+    # Calculate summed (and individual mesh) strengths for each photon group
+    summed_strengths = []
+    strengths_list = []
+    for group in range(photon_groups):
+        total_strengths = np.sum(sd[:, group])  # Sum over all mesh elements
+        summed_strengths.append(total_strengths)
+        strengths = sd[:,group]
+        strengths_list.append(strengths)
```
I recommend putting this in a method that does everything you need for a single file.  That kind of modularity enhances readability.  You might also want to separate the functions of reading from a file, processing the data, and writing to files, so methods like `extract_source_data`, `arrange_data`, `write_source_density`

gonuke commented on Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,41 @@
+            source.write(' '.join(map(str, tet_element)) + '\n')
+
+    # Calculate summed (and individual mesh) strengths for each photon group
+    summed_strengths = []
+    strengths_list = []
```
make these numpy arrays to start with?

gonuke commented on Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,41 @@
+    summed_strengths = []
+    strengths_list = []
+    for group in range(photon_groups):
+        total_strengths = np.sum(sd[:, group])  # Sum over all mesh elements
+        summed_strengths.append(total_strengths)
+        strengths = sd[:,group]
+        strengths_list.append(strengths)
```
How is `strengths_list` different from `sd`?  `sd` is a 2-D numpy array of size (`num_elements`, `num_groups`).  It looks like `strengths_list` is a list of lists with `num_group` lists, each of length `num_elements`.

Also - you don't appear to use any of the variables made in this loop?

anu1217 replied:
Resolving as this method is no longer in use

gonuke commented on Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,32 @@
+import matplotlib.pyplot as plt
+
+with openmc.StatePoint("statepoint.10.h5") as sp:
+    flux_spectrum = sp.get_tally(id=2) 
+    mesh=sp.meshes[1]
+    # Get the reshaped tally data
+    tally_data_reshaped = flux_spectrum.get_reshaped_data(value='mean')
+
+    flux_sum_en = tally_data_reshaped.sum(axis=(0,1,3,4,5))   
+    #Vitamin-J energy filter:
+    e_filter = flux_spectrum.filters[2]
+    #Lower bounds of the energy bins
+    e_filter_lower = e_filter.bins[:, 0]
```
A lot of magic numbers here including tally ID, mesh number, axes for sum, etc...  more readable if these are named as variables, and might make sense if they lived in a function that took some of these as arguments.

gonuke commented on Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,32 @@
+    #Lower bounds of the energy bins
+    e_filter_lower = e_filter.bins[:, 0]
+
+    # Plot flux spectrum
+    fix, ax = plt.subplots()
+    ax.loglog(e_filter_lower, flux_sum_en, drawstyle='steps-post')
+    ax.set_xlabel('Energy [eV]')
+    ax.set_ylabel('Flux [photon-cm/source]')
+    ax.grid(True, which='both')
+    plt.savefig('Photon_flux_vs_energy.png')
+    plt.show()
+    
+    mesh_data = tally_data_reshaped.sum(axis=(0,2,3,4,5))
+    vtk = mesh.write_data_to_vtk(filename="Photon_Flux.vtk", datasets={"mean":mesh_data.flatten()})
```
separation of concerns: make different functions/methods for reading data, processing data, writing data, and plotting data.

gonuke reviewed (changes requested):
A number of suggestions here.  We should meet to discuss more.

I think all of these files should be in the `WC_Layers` directory, and the neutron transport file should be updated to use the same geometry and materials scripts.

gonuke commented on OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,137 @@
+# Exporting materials, geometry, and tallies to .xml
+def export_to_xml(filepath, element_1, element_2, inner_radius_W, outer_radius_W, inner_radius_C):
+    OpenMC_SF = mat_lib(filepath)
+    OpenMC_W = make_W(element_1, OpenMC_SF)
+    OpenMC_C = make_C(element_2, OpenMC_SF)
```
```suggestion
    materials = []
    for material_id, element in enumerate(elements):
         materials.append(make_element(element, material_id+1, OpenMC_SF))
```

anu1217 committed:
Update OpenMC_PhotonTransport.py

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Update TwoLayers_Materials.py

- Created dictionary with material densities
- Created OpenMC Material and Materials objects with separate function

gonuke commented on TwoLayers_Materials.py:
```diff
@@ -0,0 +1,32 @@
+        mat.add_element(element, 1.00)
+        mat.set_density('g/cm3', density_dict.get(f'{element}'))
+        mats.append(mat)
+    return mats
```
To elaborate on my previous suggestion, I think it makes sense to process the data into a useful form as you read it.

It then makes it a little simpler to use it later on.

```suggestion
def alara_element_densities(filepath):  
    with open(""+filepath+"") as ALARA_Lib:
        libLines = ALARA_lib.readlines()
    num_lines = len(libLines)
    density_dict = {}
    line_num = 0
    while line_num < num_lines:
        element_data = line[line_num].strip().split()
        element_name = element_data[0].lower
        density_dict[element_name] = float(element_data[3])
        line_num += int(element_data[4]) + 1
    return density_dict
    
# Create materials & export to XML:
#Simulating tungsten shell:
def make_element(elements, density_dict):
    mats = openmc.Materials([])
    for element_id, element in enumerate(elements):
        mat = openmc.Material(material_id=element_id+1, name=element)
        mat.add_element(element, 1.00)
        mat.set_density('g/cm3', density_dict.get(element))
        mats.append(mat)
    return mats

```

gonuke commented on TwoLayers_Geometry.py:
```diff
@@ -0,0 +1,33 @@
+
+# Create geometry
+#Spherical shell:
+def make_spherical_shell(inner_radius_W, outer_radius_W, inner_radius_C, M_1, M_2):    
```
This makes multiple shells, so maybe change the name.

Could also generalize this to take an inner radius plus layers:
```suggestion
def make_spherical_shells(inner_radius_W, layers):    
    '''
    Builds a set of nested spherical shells, with an inner_radius and
    multiple layers each with a material definition and a thickness
    
    inputs
    =====
    inner_radius : the radius of the inner surface of the first shell
    layers : list of tuples where each tuple is (material, thickness)
    '''
```

anu1217 replied:
I think materials will need to be an input as this function is also assigning materials to the geometry. I'm wondering if this can be done with one set of radii/thickness inputs and the materials from TwoLayers_Materials.

```suggestion
def make_spherical_shell(radii, materials):
    '''
    Takes a list of materials and radii (outermost to innermost)
    and creates an OpenMC Geometry object.
    
    element_list = list of elements (str) in order from radially outermost to innermost
    radii = list of the radii (outermost to innermost) that serve as the boundaries of the 
    spherical shells; must be in the same order as the specified materials
    '''
    cells = []
    for index, material in enumerate(materials):
        outer_sphere = openmc.Sphere(r = radii[index]) # add boundary_type here
        inner_sphere = openmc.Sphere(r = radii[index + 1]) 
        element_region = +inner_sphere & -outer_sphere
        element_shell = openmc.Cell(fill = materials[index], region = element_region)
        cells.append(element_shell)
        if material == materials[-1]:
            void = openmc.Cell(fill = None, region = -inner_sphere) 
            cells.append(void)
    geometry = openmc.Geometry(cells)  
    return geometry    

```

gonuke replied:
I agree that the materials are necessary.  My version had them bound in the `layers`.  This kind of data structure can lead to clarity because the data gets bound together in instructive ways.  An example of calling it might be:
```
inner_radius = 10.0
# material_a and material_b are OpenMC Materials
layers = [(material_a, 5.0), (material_b, 4.0)]
geometry = make_spherical_shells(inner_radius, layers)
```

The function may be:
```
def make_spherical_shells(inner_radius, layers):

    inner_sphere = openmc.Sphere(inner_radius)
    cells = [openmc.Cell(region=-inner_sphere)]
    for (material, thickness) in layers:
        outer_radius = inner_radius + thickness
        outer_sphere = openmc.Sphere(outer_radius)
        cells.append(openmc.Cell(fill = material, region = inner_sphere & -outer_sphere))
        inner_radius = outer_radius
        inner_sphere = outer_sphere
    cells.append(openmc.Cell(region = outer_sphere)
    geometry = openmc.Geometry(cells)

    return geometry
```

(There may be minor syntax errors, but this is the basic idea)

gonuke reviewed (changes requested):
More suggestions as you chip away at this

anu1217 committed:
Update TwoLayers_Materials.py

Make dictionary of densities with ALARA element library

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Update references to some variables

anu1217 committed:
Update TwoLayers_Geometry.py

Redefine inner and outer W spheres

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Import variables from TwoLayers_Materials

anu1217 committed:
Update directory of OpenMC_PhotonTransport

anu1217 committed:
Update directory of Photon_Tallytovtk

anu1217 committed:
Update directory of TwoLayers_Geometry

anu1217 committed:
Update directory of TwoLayers_Materials

anu1217 committed:
Update directory of Source_Mesh_Reader

anu1217 committed:
Update TwoLayers_Geometry.py

- Redefined make_spherical_shells to take layers and inner_radius as input
- Deleted reference to TwoLayers_Materials to maintain generality

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
A few requests for this file:
* add argparse for filenames (with defaults)
* add separate functions to:
    * read data from statepoint files
    * calculate summed strengths
    * write data to files
* all of those should be called from a main function

gonuke commented on WC_Layers/Photon_TallytoVtk.py:
Some requests for this file:
* add argparse for filename (with defaults)
* create separate functions to:
    * read data from statepoint
    * plot data (optoinal via argparse)
    * write data to file (filename set by argparse)
* all of these called from a main function

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,138 @@
+    for material_id, element in enumerate(elements):
+         materials.append(make_element(element, material_id+1, OpenMC_SF))
+    OpenMC_Mat = all_mat(OpenMC_W, OpenMC_C)
+    OpenMC_Geometry = make_spherical_shell(inner_radius_W, outer_radius_W, inner_radius_C, OpenMC_W, OpenMC_C)
```
update for new function name and signature

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,138 @@
+    spectrum_tally.scores = ['flux']
+    
+    tall = openmc.Tallies([neutron_tally, spectrum_tally])
+    tall.export_to_xml()
```
don't export in each function, just return the OpenMC tallies object for handling by the calling function

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,138 @@
+    return universe_ss
+
+# Exporting materials, geometry, and tallies to .xml
+def export_to_xml(filepath, element_1, element_2, inner_radius_W, outer_radius_W, inner_radius_C):
```
This function is doing too much - it should probably just be making an OpenMC model out of:
* materials
* geometry
* settings (incl. sourc)
* tallies

and returning that model

gonuke replied:
I'd also recommend using lowercase/snakecase variable names

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,138 @@
+
+Mesh_File = 'OpenMC_Mesh.h5m'
+
+parser = argparse.ArgumentParser(description="Specify required inputs: file path to ALARA Element Library, element name, inner radius [cm], outer_radius [cm]")
+
+#Required user inputs:
+parser.add_argument('--filepath', type=str, required=True)    
+parser.add_argument('--element_1', type=str, required=True)
+parser.add_argument('--element_2', type=str, required=True)
+#Shell radii are left as user inputs, but the mesh is specific to W_inner_radius = 1000, W_outer_radius = 1005, C_inner_radius = 995
+parser.add_argument('--W_inner_radius', type=float, required=True)
+parser.add_argument('--W_outer_radius', type=float, required=True)
+parser.add_argument('--C_inner_radius', type=float, required=True)
+
+args = parser.parse_args()
```
Put this in a function

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,138 @@
+from TwoLayers_Materials import *
+from TwoLayers_Geometry import *
+
+Mesh_File = 'OpenMC_Mesh.h5m'
```
set this with `argparse`, possibly with default

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,138 @@
+
+args = parser.parse_args()
+
+fp = args.filepath
+E_1 = args.element_1
+E_2 = args.element_2
+R_W_1 = args.W_inner_radius
+R_W_2 = args.W_outer_radius
+R_C_1 = args.C_inner_radius
```
Put this in a main function

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,138 @@
+    return Source_List, Particle_Filter, Total_Mesh, Mesh_Dist
+
+# Define tallies
+def tallies(W_Shell, C_Shell, Particle_Filter, Total_Mesh):
```
given the flexibility of the method to make cells, this may make more sense if it took a list of cells

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,138 @@
+#Define source:
+def make_source(W_Shell, C_Shell, Cells):
+    Source_List = []
+    Particle_Filter = openmc.ParticleFilter('photon')
```
should this be added in tallies?

gonuke reviewed (changes requested):
A few more high level comments here.  They will probably result in a lot of code changes, but shouldn't take too long.  Conceptually, it's a lot of same changes across many files.

anu1217 committed:
Add functions to read, plot, and save data

anu1217 committed:
Fix indentation of return statement

anu1217 committed:
Created functions to extract and write relevant data

anu1217 committed:
Define arrange_source_strengths function

- The previous version was not storing source strengths correctly. The data from the last entry in sd_list was overwriting the data from the previous entries. This has been fixed in this commit.

anu1217 committed:
Add main function that calls all helper functions

anu1217 committed:
Rewrite make_source in terms of data_extractor from Source_Mesh_Reader

anu1217 committed:
Fix summation in make_source loop

anu1217 committed:
Changed variable name in function definition

anu1217 committed:
Updated main function that calls helper functions

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,64 @@
+def extract_source_data(file_list):
+    '''
+    Identifies the location of the source density dataset within each mesh file.
+    '''
```
It is useful to document the input and output information in the docstring - here and elsewhere .  For example, it looks like the input expects a list of h5py-opened files (which may not be obvious) and the output is even less clear.  I would also recommend that this function take a list of filenames as input and take care of opening/closing the files within the function.

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,64 @@
+File_2 = h5py.File("source_mesh_2.h5m", 'r')
+Files = [File_1, File_2]
+
+def extract_source_data(file_list):
```
Great addition!

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,64 @@
+    sd_list = []
+    for file in file_list:
+        # Extract data from the current file
+        tstt = file['tstt']
+        elements = tstt['elements']
+        tet_four = elements['Tet4']
+        tags = tet_four['tags']
+        sd = tags['source_density'][:]
+        sd_list.append(sd)
```
Since you don't actually use any of these intermediate variables, I think it's probably OK do say:
```suggestion
        sd_list.append(file['tstt']['elements']['Tet4']['tags']['source_density'][:])
```

gonuke replied:
I have a feeling that it may also be possible to use a path syntax like:
`"tsst/elements/Tet4/tags/source_density"`, but I'm not sure.

anu1217 replied:
I gave this a quick try and it didn't seem to work

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,64 @@
+    
+def arrange_source_strengths(sd_list, photon_groups):
+    '''
+    Writes individual mesh element strengths (separated by energy group) to density_list, 
```
It doesn't look like this method "writes" anything...  It may not be useful to describe the steps that are shown in the code, but summarize the overall purpose.

anu1217 replied:
I think this function could be deleted entirely - the source strengths and the relevant sums are now being accessed directly through extract_source_data in the photon transport code

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,64 @@
+    then writes density_list to all_strengths_list for each set of source densities. 
+    '''
+    all_strengths_list = []
+    for density in sd_list:
+        density_list = []
+        # Calculate (individual mesh) strengths for each photon group
+        for group in range(photon_groups):
+            strengths = density[:,group]
+            density_list.append(strengths)
+        all_strengths_list.append(density_list)    
```
I am not sure I understand the shape of these data structures, but looking at the operations, it seems like this could be one or two lines using numpy operations.  I recommend that all data be treated as numpy arrays as soon as possible

gonuke replied:
If I read this correctly, you are looping over the data from each file (i.e. each shutdown time), each of which is size (num_mesh_elements, num_groups) and making a list that is (num_groups, num_mesh_elements).  That is, you are just transposing this data.  If it was a numpy array of size `(num_files, num_mesh_elements, num_groups)` this would be one line - something like:
```
np.transpose(sd_list, (0,2,1))
```

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,64 @@
+    '''
+    Identifies the location of the source density dataset within each mesh file.
+    '''
+    sd_list = []
```
If there is a way to know the size of this data structure ahead of time, I recommend making this a numpy array that you fill with values below.  For example:
```
# assume we know num_mesh_elements, num_groups
sd_list = np.ndarray( (len(file_list), num_mesh_elements, num_groups) )

for file_num, file in enumerate(file_list):
    sd_list[:, file_num] =  file['tstt']['elements']['Tet4']['tags']['source_density'][:]
```

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,64 @@
+        # Write source density to a separate file for each mesh
+        with open(f'{sd_filename}_{sd_index + 1}.txt', 'w') as source:
+            for tet_element in source_density:
+                source.write(' '.join(map(str, tet_element)) + '\n')
```
is this joining across groups?  Are you writing a multi-group source for each mesh? Or would it make more sense to write `num_groups` separate files for each mesh?

gonuke reviewed (changes requested):
A few questions about how the source mesh reader works

anu1217 committed:
Rename write_source_density

anu1217 committed:
Rewrite extract_source_data

anu1217 committed:
Change extract_source_data docstring

anu1217 committed:
Moved Particle_Filter to tallies

anu1217 committed:
Add inner_radius as input

anu1217 committed:
Change inputs required for make_source

anu1217 committed:
Remove Mesh_Dist as output of make_source

anu1217 committed:
Change inputs required for tallies & fix capitalization

anu1217 committed:
Change variable names from uppercase to lowercase

anu1217 committed:
Rewrite export_to_xml as create_openmc_model

- create_openmc_model creates a model object using all other OpenMC objects

- deleted plot_universe for now as this function cannot create an OpenMC plots object

- deleted .export_to_xml calls

anu1217 committed:
Deleted comments describing self-explanatory code

anu1217 committed:
Add docstrings to functions

anu1217 committed:
Add input to tallies function

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,47 @@
+
+def save_source_density(sd_list, sd_filename):
+    '''
+    Saves source density data as a text file.
```
It looks like you open multiple files.  Can you add more to the docstring about the difference between files and the structure of the files?

gonuke reviewed (changes requested):
I added one  additional request about a docstring.  

I see you chipping away at things in this PR - can you perhaps move it to "DRAFT" until it's ready for another complete review.

anu1217 committed:
Changed bounds to numpy array

- Modified loop over bounds in make_source (no longer uses enumerate)

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,142 @@
+    '''
+    source_list = []
+    total_mesh = openmc.UnstructuredMesh(mesh_file, library='moab')
+    for index in range(len(bounds) - 1):
+        mesh_dist = openmc.stats.MeshSpatial(total_mesh, strengths=esd[mesh_index][:,index], volume_normalized=False)
+        energy_dist = openmc.stats.Uniform(a=bounds[index], b=bounds[index + 1])
+        source_list.append(openmc.IndependentSource(space=mesh_dist, energy=energy_dist, strength=np.sum(esd[mesh_index][:, index]), particle='photon', domains=cells))
```
```suggestion
    for index, (lower_bound, upper_bound) in enumerate(zip(bounds[:-1],bounds[1:[)):
        mesh_dist = openmc.stats.MeshSpatial(total_mesh, strengths=esd[mesh_index][:,index], volume_normalized=False)
        energy_dist = openmc.stats.Uniform(a=lower_bound, b=upper_bound)
        source_list.append(openmc.IndependentSource(space=mesh_dist, energy=energy_dist, strength=np.sum(esd[mesh_index][:, index]), particle='photon', domains=cells))
```

gonuke reviewed (changes requested):
Here is a little pythonic way to get lower and upper bounds in a loop

anu1217 committed:
Change the way upper & lower energy bounds are accessed

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Fixed typo in make_source

anu1217 committed:
Add inputs to yaml file

anu1217 committed:
Update and rename PhotonTransport_Inputs.yaml to WC_Layers/PhotonTransport_Inputs.yaml

anu1217 committed:
Remove hard-coded inputs that have been moved to yaml

anu1217 commented on WC_Layers/TwoLayers_Geometry.py:
```diff
@@ -0,0 +1,30 @@
+
+# Create geometry
+#Spherical shell:
+def make_spherical_shells(layers, inner_radius):    
+    '''
+    Creates a set of concentric spherical shells, each with its own material & inner/outer radius.
+    
+    layers: list of tuples with OpenMC Material name and thickness: (material, thickness)
+    inner_radius: the radius of the innermost spherical shell
+    '''
```
I wonder if this function could take materials and thicknesses separately, and make layers within the function. This would avoid having the user provide a list of tuples beforehand

```suggestion
def make_spherical_shells(materials, thicknesses, inner_radius):    
    '''
    Creates a set of concentric spherical shells, each with its own material & inner/outer radius.
    
    layers: list of tuples with OpenMC Material name and thickness: (material, thickness)
    inner_radius: the radius of the innermost spherical shell
    '''
    layers = zip(materials, thicknesses)
```

anu1217 committed:
Merge branch 'cnerg:main' into mesh_photontransport

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,78 @@
+    outputs:
+        talls: OpenMC Tallies object
+    '''
+    particle_filter = openmc.ParticleFilter('photon')
```
You define this particle filter, which makes sense for a photon transport problem, but all the rest of your tally definition seems like it is meant for neutrons, especially the choice of scores, energy filter

anu1217 replied:
Yes, tally 1 originated from the neutron transport model. Perhaps a heating/absorption photon tally (with the appropriate variable names) would provide more relevant information?

gonuke replied:
For R2S problems, it's customary to do an equivalent dose tally.

gonuke commented on WC_Layers/Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,35 @@
+import openmc
+import matplotlib.pyplot as plt
+
+def read_statepoint(sp_filename, flux_spectrum_tally_id, mesh_number, summed_axes_energy_filter, energy_filter_index):
```
With the addiction of a docstring here, I think some better variable names may emerge:
* `flux_spectrum` => `flux_spectrum_tally`
* `tally_data_reshaped` => `flux_spectrum_mean`
* `flux_sum_energy` => `total_flux`

gonuke commented on WC_Layers/Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,35 @@
+        mesh=sp.meshes[mesh_number]
+        # Return tally data condensed into 1 dimension per filter
+        tally_data_reshaped = flux_spectrum.get_reshaped_data(value='mean')
+        flux_sum_energy = tally_data_reshaped.sum(axis=summed_axes_energy_filter)   
```
It's not clear to me what axes this data is being summed over.  The data is originally a flux spectrum of G groups for each of M mesh elements. I assumed that this sum was summing over G, to produce a total flux for each of the M mesh elements.  However, when you use this in the plot below, it seems like it may be summing over M to produce a single flux spectrum of G groups.  If so, then we need to make sure that the volume normalization is being done correctly. (Full disclosure: I'm not sure when/how volume normalization of tallies happens in OpenMC.  If this was MCNP-like, then this operation would be wrong because the results would already be volume normalized per mesh element, and summing would require a volume-weighting.)

anu1217 replied:
For the purpose of plotting the data, the summation is happening over M so that the flux is divided into G groups. 

From digging into r2s.py for a bit, I believe that the outputted photon source mesh info is normalized by volume. That said, I don't think there is any division by volume automatically happening on the OpenMC side, so ALARA/r2s may be outputting the wrong units if it is expecting values from MCNP. However, I do have the volume normalization set to False within MeshSpatial of the PhotonTransport code, so there is no remultiplication by volume of the source strengths. I think this is still worth looking into further.

gonuke reviewed (changes requested):
I'll keep commenting as I get a chance as this evolves.  Here are a few more questions/ideas.

anu1217 committed:
Add information to calculate effective dose

anu1217 committed:
Redefine inputs for make_spherical_shells

anu1217 committed:
Edit function docstring

anu1217 committed:
Changed inputs for extract_source_data

anu1217 committed:
Add main function to Source_Mesh_Reader

anu1217 committed:
Add default to argparse input

anu1217 committed:
Add files via upload

anu1217 committed:
Merge branch 'cnerg:main' into mesh_photontransport

anu1217 committed:
Add input to function

anu1217 committed:
Add docstrings to functions

anu1217 committed:
Change variable names and inputs

anu1217 committed:
Change variable key

anu1217 committed:
Update PhotonTransport_Inputs.yaml

anu1217 committed:
Fix typo in run mode

anu1217 committed:
Update and rename Source_Mesh_Reader_Inputs.yaml to WC_Layers/Source_Mesh_Reader_Inputs.yaml

anu1217 committed:
Fix variable updating to avoid seg fault

anu1217 committed:
Add main() function to compile all functions

anu1217 committed:
Calculate dose and add main() function

anu1217 committed:
Update PhotonTransport_Inputs.yaml

anu1217 committed:
Update Source_Mesh_Reader.py

anu1217 committed:
Create README.md

anu1217 committed:
Update and rename WC_Layers/Conversion_h5m.py to WC_Layers/R2S_Step1_Input/Conversion_h5m.py

anu1217 committed:
Update and rename WC_Layers/PyNE_Lib.py to WC_Layers/R2S_Step1_Input/PyNE_Lib.py

gonuke commented on WC_Layers/TwoLayers_Geometry.py:
```diff
@@ -0,0 +1,23 @@
+        thicknesses: thickness of each OpenMC Material
+        inner_radius: the radius of the innermost spherical shell
+    '''
+    layers = zip(materials, thicknesses)
```
Might be better to zip these together before calling the function (or as you call the function).  This reinforces that these need to be bound to each other in this way.

anu1217 replied:
This function now takes layers as input.

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,49 @@
+        inputs = yaml.safe_load(yaml_file)
+        
+    source_mesh_list = inputs['source_meshes']
+    num_elements = inputs['num_elements']
```
Is the number of mesh elements defined by the mesh?  Doesn't this introduce the possibility that there is a discrepancy introduced by the user?

anu1217 replied:
We set it up this way so that `sd_list` could be a numpy array instead of a list. Specifying the wrong value of either the number of mesh elements or photon groups will produce an error

gonuke replied:
can't you query the mesh for this data, though? instead of having the user specify it?

anu1217 replied:
The mesh size is now read from one of the source density datasets.

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,49 @@
+    sd_list = np.ndarray((len(source_mesh_list), num_elements, photon_groups))
+    for source_index, source_name in enumerate(source_mesh_list):
+         file = h5py.File(source_name, 'r')
+         sd_list[source_index,:] = file['tstt']['elements']['Tet4']['tags']['source_density'][:]
```
`sd_list` is initialized as three-dimensional, but you only refer to 2 dimensions here.  That may be OK if both colons refer to 2-D data?

anu1217 replied:
Yes, both colons refer to the 2D photon_groups X num_elements data. This is just making a list of the 2D data for each decay step

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,49 @@
+    sd_list = np.ndarray((len(source_mesh_list), num_elements, photon_groups))
+    for source_index, source_name in enumerate(source_mesh_list):
+         file = h5py.File(source_name, 'r')
+         sd_list[source_index,:] = file['tstt']['elements']['Tet4']['tags']['source_density'][:]
```
Probably good to have a comment before this:
```suggestion
         # directly accessing the data stored in the HDF5 as specified by the MOAB mesh library
         # MOAB data is stored in the `tstt` group, and this data is in a `tag` with name `source_density`
         # applied to each of the `Tet4` type of `elements`
         sd_list[source_index,:] = file['tstt']['elements']['Tet4']['tags']['source_density'][:]
```

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,49 @@
+        
+    source_mesh_list = inputs['source_meshes']
+    num_elements = inputs['num_elements']
+    photon_groups = inputs['photon_groups']
```
Isn't the number of groups already defined internally to the mesh file?

anu1217 replied:
Same as mentioned above regarding the number of mesh elements

gonuke commented on WC_Layers/README.md:
Might want to review the markdown formatting to see if there's a better way to format this file

gonuke commented on WC_Layers/Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,94 @@
+import argparse
+import numpy as np
+
+def read_statepoint(sp_filename, flux_spectrum_tally_id, summed_axes_energy_filter, energy_filter_index, mesh_number):
```
You seem to do most of the same things as this in your other problem/project.  I wonder if you can reuse it in someway?

gonuke commented on WC_Layers/Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,94 @@
+        e_filter_lower = e_filter.bins[:, 0]
+    return e_filter_lower, total_flux, flux_spectrum_mean, mesh
+
+def plot_flux_data(e_filter_lower, total_flux, figure_filename):
```
This separation of concerns is better than in the other project with a single spherical shell! Great! 👍

gonuke commented on WC_Layers/Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,94 @@
+        summed_axes_dose_filter: n-tuple of axes over which tally data is summed in order to isolate dose filter axis
+    '''    
+    total_dose = flux_spectrum_mean.sum(axis=eval(summed_axes_dose_filter))
+    return total_dose
```
While I'm often a fan of short methods that have clear purposes due to their name, this seems like it could be clear if done inline.

gonuke commented on WC_Layers/Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,94 @@
+    plt.savefig(figure_filename)
+    plt.show()
+
+def calculate_dose(flux_spectrum_mean, summed_axes_dose_filter):
```
If the result of this is the `total_dose` as just as a sum of the input, then it probably means the input isn't a flux spectrum.

anu1217 replied:
No, the input isn't a flux spectrum -- it's just that the flux is a spectrum on this tally, so it's summed over the relevant dimensions to extract the total dose

gonuke commented on WC_Layers/Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,94 @@
+    total_dose = flux_spectrum_mean.sum(axis=eval(summed_axes_dose_filter))
+    return total_dose
+
+def write_to_vtk(flux_spectrum_mean, summed_axes_mesh_filter, vtk_filename, mesh):
```
Either this method name should to change to indicate that you're summing over some dimensions, or the sum should happen before you call the method.  The fact that it's summed is obscured in this case.

gonuke commented on WC_Layers/Photon_TallytoVtk.py:
```diff
@@ -0,0 +1,94 @@
+    summed_axes_dose_filter = inputs['summed_axes_dose_filter']
+    cd = calculate_dose(rs[2], summed_axes_dose_filter)
+    
+    summed_axes_mesh_filter = inputs['summed_axes_mesh_filter']
```
Looking at the YAML, the meaning/interpretation of this is not very obvious/clear.

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,136 @@
+    total_filter = openmc.MeshFilter(total_mesh)
+    
+    photon_tally = openmc.Tally(tally_id=1, name="Photon tally")
+    photon_tally.scores = ['flux', 'elastic', 'absorption']
```
Are these scores relevant/interesting for your photon tally?

gonuke replied:
Probably want the `bounds` from your YAML file

anu1217 replied:
I'm getting rid of the elastic tally as it is 0, but it might be interesting to see how the flux and absorption spectra evolve relative to each other (once `bounds` is applied as an energy filter)

anu1217 replied:
`photon_tally` is now set up to record flux and absorption over `bounds`, but I'm wondering if the absorption score can just be moved to `spectrum_tally` instead

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,136 @@
+        OpenMC Tallies object
+    '''
+    particle_filter = openmc.ParticleFilter('photon')
+    total_filter = openmc.MeshFilter(total_mesh)
```
I'm not sure what `total` means in these variable names?

anu1217 replied:
I meant 'total' as in the unstructured mesh encompasses the full geometry, but I realize that's not clear from the variable name. Renamed `total_mesh` as `unstructured_mesh`.

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,136 @@
+    dose_filter = openmc.EnergyFunctionFilter(dose_energy, dose)
+
+    # An energy filter is created to assign to the flux tally.
+    energy_filter_flux = openmc.EnergyFilter.from_group_structure("VITAMIN-J-42")
```
I'm not sure this is relevant for photons??

anu1217 replied:
I thought the 42-group structure was for photons and the 175 one for neutrons...or should this change to match the group structure from ALARA?

gonuke replied:
My bad... I didn't properly see the "42".  I think you are right.  It may also be useful to have the same one as ALARA uses at some point.

gonuke commented on WC_Layers/OpenMC_PhotonTransport.py:
```diff
@@ -0,0 +1,136 @@
+
+    spectrum_tally = openmc.Tally(tally_id=2, name="Flux spectrum")
+    # Implementing energy and cell filters for flux spectrum tally
+    spectrum_tally.filters = [cell_filter, total_filter, energy_filter_flux, particle_filter, dose_filter]
```
maybe you should apply the `dose_filter` to the `photon_tally` instead of the `spectrum_tally`?

anu1217 replied:
If photon_tally is changed to include an energy filter with the ALARA bounds, I'm not sure it would matter which tally the dose filter is associated with

gonuke reviewed (changes requested):
I didn't note all the places that comments from other PR's/projects might be applied here...

anu1217 committed:
Modularize main() and change total_mesh to unstructured_mesh

anu1217 committed:
Modularize main()

anu1217 committed:
Modularize main()

anu1217 committed:
Change inputs for make_spherical_shells()

anu1217 committed:
Update WC_Layers/Source_Mesh_Reader.py

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Change variable names and function inputs

anu1217 committed:
Change variable names and function inputs

anu1217 committed:
Add extra parameters

gonuke commented on WC_Layers/TwoLayers_Geometry.py:
```diff
@@ -0,0 +1,21 @@
+import openmc
+
+def make_spherical_shells(layers, inner_radius, outer_boundary_type):    
```
very minor point: for me it's more logical to start with the `inner_radius` because then your function arguments come in the same order as the geometry.

gonuke commented on WC_Layers/Source_Mesh_Reader.py:
```diff
@@ -0,0 +1,58 @@
+                source.write(' '.join(map(str, tet_element)) + '\n')
+
+def main():
+    def parse_args():
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--Mesh_Reader_YAML', default = 'Source_Mesh_Reader_Inputs.yaml', help="Path (str) to YAML file containing required inputs for Source_Mesh_Reader")
+        args = parser.parse_args()
+        return args
+    
+    def read_yaml(args): 
+        with open(args.Mesh_Reader_YAML, 'r') as yaml_file:
+            mesh_reader_inputs = yaml.safe_load(yaml_file)
+        return mesh_reader_inputs
+
+    def read_source_mesh(mesh_reader_inputs):
+        source_data = extract_source_data(mesh_reader_inputs['source_meshes'],
+                                      mesh_reader_inputs['num_elements'],
+                                      mesh_reader_inputs['photon_groups'])
+        save_source_density(source_data, 
+                              mesh_reader_inputs['sd_filename'])
```
nested functions probably not appropriate here

gonuke reviewed (changes requested):
A few more comments here, but it may be time for another refactor to simplify things somewhat.

anu1217 committed:
Change order of inputs

anu1217 committed:
Change order of inputs to make_spherical_shells()

anu1217 committed:
Fix typo

anu1217 committed:
Modularize main()

anu1217 committed:
Remove num_elements and num_photon_groups from user-specified quantities

anu1217 committed:
Remove num_elements and photon_groups from YAML input

anu1217 committed:
Remove num_elements and photon_groups from user-defined inputs

anu1217 commented:
I believe I've addressed all the feedback - the only other thing I think is helpful would be to condense Source_Mesh_Reader and OpenMC_PhotonTransport into one file, but I'm also not sure if this would violate separation of concerns. I think TwoLayers_Geometry and TwoLayers_Materials should remain separated since the functions from the two files get imported into various other input files (photon transport, neutron transport, and OpenMC-only R2S once it's PR'ed)

anu1217 committed:
Modularize main()

gonuke commented:
This is definitely looking better, but I realize that the already merged neutron transport step in the `WC_Layers` folder has fallen out of date with these improvements.  Specifically, you don't import the geometry or materials information into the neutron transport step yet.  Recall, also, some earlier discussions that separation of concerns doesn't require different files, but really focuses on different functions that could live in the same file.

Ultimately, I think many R2S problems could have a single OpenMC file that contains method(s) to describe:
* material info 
* geometry info 
* neutron tallies
* neutron source definition
* neutron settings
* photon tallies
* photon source definition, including
    * necessary methods to photon mesh source files
* photon settings

and then one method each to:
* build a neutron problem
* build a photon problem

and finally, a main method that uses argparse to decide whether to run a neutron or photon problem, from a common set of YAML data

anu1217 committed:
Merge branch 'cnerg:main' into mesh_photontransport

anu1217 committed:
Add files via upload

anu1217 committed:
Update and rename OpenMC-to-ALARA_R2S.py to WC_Layers/OpenMC-to-ALARA_R2S.py

anu1217 committed:
Update and rename OpenMC_ALARA_WC.yaml to WC_Layers/OpenMC_ALARA_WC.yaml

anu1217 commented:
I added OpenMC-to-ALARA_R2S.py, which contains all the information needed to build a neutron and/or photon transport model. The associated YAML is OpenMC_ALARA_WC.yaml.

If these are in a suitable format, I think we're good to delete the individual transport files and their respective YAMLs.

anu1217 committed:
Delete comment

anu1217 committed:
Change variable names and order of reference

gonuke commented on WC_Layers/OpenMC-to-ALARA_R2S.py:
```diff
@@ -0,0 +1,269 @@
+    geometry = openmc.Geometry(cells)    
+    return geometry
+
+def make_source(energy):
```
```suggestion
def make_neutron_source(energy):

```

gonuke commented on WC_Layers/OpenMC-to-ALARA_R2S.py:
```diff
@@ -0,0 +1,269 @@
+    
+    return openmc.Tallies([neutron_tally, spectrum_tally])
+
+def extract_source_data(source_mesh_list, num_elements, num_photon_groups):
```
```suggestion
def extract_photon_source_data(source_mesh_list, num_elements, num_photon_groups):

```

gonuke commented on WC_Layers/OpenMC-to-ALARA_R2S.py:
```diff
@@ -0,0 +1,269 @@
+def create_neutron_model(inputs, materials, geometry):
+    settings_info = inputs['settings_info']
+    
+    source = make_source(inputs['particle_energy'])
```
```suggestion
    source = make_neutron_source(inputs['particle_energy'])

```

gonuke commented on WC_Layers/OpenMC-to-ALARA_R2S.py:
```diff
@@ -0,0 +1,269 @@
+def read_source_mesh(inputs):
+    #Find the size of the first source density dataset (assumed to be the same for all other datasets):
+    sd_data = h5py.File(inputs['source_meshes'][0], 'r')['tstt']['elements']['Tet4']['tags']['source_density'][:]
+    sd_list = extract_source_data(inputs['source_meshes'],
```
```suggestion
    sd_list = extract_photon_source_data(inputs['source_meshes'],

```

gonuke reviewed (approved):
This looks great.  I've suggested renaming a couple of methods to clarify the neutron-photon separation.

Looking forward to seeing it without the extra files that are no longer necessary

anu1217 committed:
Update WC_Layers/OpenMC-to-ALARA_R2S.py

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Update WC_Layers/OpenMC-to-ALARA_R2S.py

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Delete WC_Layers/OpenMC_PhotonTransport.py

anu1217 committed:
Delete WC_Layers/PhotonTransport_Inputs.yaml

anu1217 committed:
Delete WC_Layers/Source_Mesh_Reader.py

anu1217 committed:
Delete WC_Layers/TwoLayers_Geometry.py

anu1217 committed:
Delete WC_Layers/TwoLayers_Materials.py

anu1217 committed:
Delete WC_Layers/Source_Mesh_Reader_Inputs.yaml

anu1217 committed:
Update WC_Layers/OpenMC-to-ALARA_R2S.py

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Update WC_Layers/OpenMC-to-ALARA_R2S.py

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

anu1217 committed:
Delete WC_Layers/OpenMC_xml_WC.py

gonuke reviewed (approved):
This all looks clear and compact now - thanks @anu1217
